libs/src/api/file.js

```diff
@@ -110,6 +112,24 @@ class File extends Base {
     }
   }
 
+  /**
+   * Checks if a cid exists in an IPFS node. Public gateways are tried first, then
+   * fallback to a specified gateway and then to the default gateway.
+   * @param {string} cid IPFS content identifier
+   * @param {Array<string>} creatorNodeGateways fallback ipfs gateways from creator nodes
+   */
+  async checkCID (cid, creatorNodeGateways) {
+    const gateways = creatorNodeGateways.concat(publicGateways)
+    return Promise.all(gateways.map(async (gateway) => {
+      try {
+        const { status } = await axios({ url: urlJoin(gateway, cid), method: 'head' })
+        return [gateway, status === 200]
```

Comment:
since this is naturally a key value data structure doesn't make sense to use nested arrays. a hash seems more appropriate